src/browserlib/extract-dfns.mjs

```diff
@@ -140,6 +206,18 @@ export default function (spec, idToHeading = {}) {
   }
 
   return [...document.querySelectorAll(definitionsSelector)]
+    .map(node => {
+      // 2021-06-21: Temporary preprocessing of invalid "idl" dfn type (used for
+      // internal slots) while fix for https://github.com/w3c/respec/issues/3644
+      // propagates to all EDs and /TR specs. To be dropped once crawls no
+      // longer produce warnings.
+      if (node.getAttribute('data-dfn-type') === 'idl') {
+        node.setAttribute('data-dfn-type', 'attribute');
+        console.warn('[reffy]', `Fixed invalid "idl" dfn type "${normalize(node.textContent)}"`);
+      }
+      return node;
+    })
+    .filter(hasValidType)
```

Comment:
Thinking some more - instead of removing definitions with an invalid type, shouldn't we default them back to `dfn`?

Reply:
I'm hesitant to do that. The good thing about removing definitions with an invalid type is that no one can reference them. When someone needs to do that, they will notice that the dfn is missing and raise an issue somewhere to have that fixed, and this should eventually appear on our radar.

If we default these definitions to `dfn`, people may just ask for a `data-export` attribute to be added, and then they can start to link to it, probably using a wrong shorthand syntac. We may not notice the issue until both the source spec that defines the term and the spec that references it are wrong.

Obviously, we could notice if we had a proper way to deal with the "warnings" that Reffy logs. I'm not utterly confident that we'll look into the workflow logs, and it's more work to have a better alert mechanism. Useful work but more work... Plus, if we do that, we can just as well remove the dfns: we would have them fixed in the source spec.

Reply:
ok, I can see that making kind of sense; the risks are probably limited in any case if [ReSpec starts flagging unknown definition types](https://github.com/w3c/respec/pull/3646#issuecomment-865003774) the same way bikeshed does.